### PR TITLE
Implement NGAC abstraction

### PR DESCRIPTION
**Pull request by ivario123**
# Implement NGAC abstraction and a robust method of testing them
Aims to partially solve #12 and discard the remainder of it.
Aims to partially solve #4 and leave future work to be done in NGAC branch.

Moreover this PR introduces the base API that will be used in the future for the entire implementation. Thus I think that it would be beneficiall to us all if the API we end up with is as clean as possible.

## Status of pull request

- [x] Runnable ngac servers in a "platform agnostic" manner
- [x] Loading and switching between policies
- [x] Combining policies
- [x] Allowing for static attributes and validation of access requests
- [ ] Code review

### Abstraction
I wrote a NGAC class that handles the interfacing with the server leaving the end user with a clean API akin to
```python
from ngac import NGAC
from ngac_types.ngac_policy import Policy

# Create the ngac API class
ngac = NGAC()

# Create a policy, we will load from a file on disk
pol = Policy(
        name="Vehicle Ownership Policy", path="EXAMPLES/policy_vehicle_ownership.pl"
)

# Switch to the policy
unwrap(ngac.switch_to(pol,token = "admin_token"))

# Ensure the policy is loaded
print(unwrap(ngac.get(Policy,token = "admin_token")))
```

### Improvements
I removed all the python code that I had written to start the servers since that was just bloat. The frontend should not handle the backend server. So the frontend now assumes that the backend is running, allowing for cleaner and shorter code. 

## How did you test your solution?

I wrote 5 [tests](https://github.com/ivario123/NGAC_ABE/blob/ngac_boilerplate/src/NGAC/test_ngac.py) for the servers and the API.

### CODE REVIEW

File: src/NGAC/ngac_types/user.py [hunk 1]
```diff
@@ -104,7 +103,11 @@ def test_create_with_attributes():
 
 def test_list_like_user():
     """
-    Test user class, list like operations
+    <<<<<<< HEAD
```
Comment (Leohemmingsson) [inline src/NGAC/ngac_types/user.py:106]: More merge stuff

**Review by Leohemmingsson — CHANGES_REQUESTED** [review]
Just found some small stuff, structure looks good!

File: src/NGAC/ngac.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+            "token": token,
+        }
+        response = requests.get(base_url, params=params)
+        print(response.text)
+        return "grant" in response.text
+
+    ##########################################################
+    #                        Getters                         #
+    ##########################################################
+    def get(self, type: NgacType, token: str) -> requests.Response:
```
[the remark this answers is not in the document]
Comment (ivario123) [reply src/NGAC/ngac.py]: Fixed this. Now it only exists for policies

File: src/NGAC/ngac_types/user.py (hunk 1 above)
Comment (ivario123) [reply src/NGAC/ngac_types/user.py]: Fixed

File: src/NGAC/ngac.py [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+        """
+        base_url = f"{self.policy_server_url}{GetPolicy()}"
+        # This is bad, we should make the user pass a token
+        params = {"token": f"{token}"}
+        return requests.get(base_url, params=params)
+
+    ##########################################################
+    #                        Setters                         #
+    ##########################################################
+    def switch_to(self, value: NgacType, token: str = "") -> requests.Response:
```
[the remark this answers is not in the document]
Comment (ivario123) [reply src/NGAC/ngac.py]: Changed to 
```python
def change_policy(self, target_policy: NgacType) -> requests.Response:
```
Not quite sure change_policy is the best name though

**Pull request merged**